### DISCUSSION TITLE
Sun2000L compatibility

### DISCUSSION THREAD

**Discussion by DanieleCar** [wlcrs/huawei_solar#570] [Connecting to the inverter]
hello , i didnt understand if the old sun2000l is compatible with this integration,  in the setup menu of the inverter there is the option for activating modbus un that rs485 port

**Comment by Roving-Ronin** [comment]
@DanieleCar refer to the Wiki that mentions this integration uses the Modbus registers from https://github.com/wlcrs/huawei_solar/files/10920652/Solar.Inverter.Modbus.Interface.Definitions.v05.pdf  Refer to page 6 for those inverters that are compatible / use those registers.

**Comment by DanieleCar** [comment]
thanks , look like my inverter is on the list :  SUN2000L-5KTL.
I buy the EW11 wifi dongle and connected to the right pin of inverter ( pin 1, 3).
From the status  menu of ew11 i see that i receive a lot of frame from the inverter , but none of them are trasmittet on the tcp server. when i try to connect the integration i see that the tcp side receive 2 frame and the modbus side trasmit these 2 frame to inverter , but look like that inverter dont respond or the ew11 cannot recognise the response frame .
i tryed with id 0 and 1 . any suggestion ?

**Comment by DanieleCar** [comment]
look like that sun2000L has the  modbus tcp server integrated in the wlan client . so i tried to connect to the inverter ip and port 502 with some success 
![immagine](https://github.com/wlcrs/huawei_solar/assets/142917031/70095a7a-9682-4f3c-8a96-b900d2bfddf5)

i also tryed with modbus pol ( a client ) and i can read register 30000 without problems 

so now the question is : why the integration canno start? i also turn on the debug 
[home-assistant_huawei_solar_2023-12-26T10-24-39.146Z.log](https://github.com/wlcrs/huawei_solar/files/13770800/home-assistant_huawei_solar_2023-12-26T10-24-39.146Z.log)

Thank you for all the work and support!!
